src/tools/immutable_object_cache/CacheClient.h

```diff
@@ -0,0 +1,55 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_CACHE_CACHE_CLIENT_H
+#define CEPH_CACHE_CACHE_CLIENT_H
+
+#include <atomic>
+#include <boost/asio.hpp>
+#include <boost/bind.hpp>
+#include <boost/asio/error.hpp>
+#include <boost/algorithm/string.hpp>
+#include "include/ceph_assert.h"
+#include "include/Context.h"
+#include "SocketCommon.h"
+
+
+using boost::asio::local::stream_protocol;
+
+namespace ceph {
+namespace immutable_obj_cache {
+
+class CacheClient {
+public:
+  CacheClient(const std::string& file, CephContext* ceph_ctx);
+  ~CacheClient();
+  void run();
+  bool is_session_work();
+
+  void close();
+  int stop();
+  int connect();
+
+  int register_client(Context* on_finish);
+  int lookup_object(std::string pool_name, std::string object_id, Context* on_finish);
```

Comment:
I think it is better to use pool ID instead of the pool name (here and in the cache store). The pool can be renamed and it does not affect opened images.

(could be a TODO if we want to merge this earlier)

Reply:
right, should also avoid to use pool name when persistent to local cache device as it main contain special characters